src/caffe/layers/prelu_layer.cu

```diff
@@ -83,13 +88,13 @@ void PReLULayer<Dtype>::Backward_gpu(const vector<Blob<Dtype>*>& top,
     Dtype* slope_diff = this->blobs_[0]->mutable_gpu_diff();
     int cdim = channels * dim;
     Dtype dsum = 0.;
-    for (int n = 0; n < bottom[0]->num(); ++n) {
+
```

Comment:
Yes, that had been true had the multiplier been variable for every N. But because it is 1 the caffe_gpu_dot is essentially doing a sum across the CxHxW and the loop is adding that sum for all the inputs in the batch.
Also for caffe_gpu_gemv the sum of HxW for every channel C is computed which is then added up for all inputs. 
So what I am doing is that for all the N (inputs) in NxCxHxW blob, I am calculating sum across N for each of the CxHxW elements. After this just a single call to caffe_gpu_dot or caffe_gpu_gemv should work.